crates/re_types/src/arrow_buffer.rs

```diff
@@ -0,0 +1,30 @@
+use arrow2::buffer::Buffer;
+
+/// Convenience-wrapper around an arrow [`Buffer`] that is known to contain a
+/// a primitive type.
+///
+/// The arrow2 [`Buffer`] object is internally reference-counted and can be
+/// easily converted back to a `&[T]` referencing the underlying storage.
+/// This avoids some of the lifetime complexities that would otherwise
+/// arise from returning a `&[T]` directly, but is significantly more
+/// performant than doing the full allocation necessary to return a `Vec<T>`.
+#[derive(Clone, Debug, Default, PartialEq)]
+pub struct ArrowBuffer<T>(pub Buffer<T>);
+
+impl<T> ArrowBuffer<T> {
+    #[inline]
+    pub fn len(&self) -> usize {
+        self.0.len()
+    }
```

Comment:
⚠️ confusing name

I think we should rename this `num_bytes` and add a separate `num_elements` method, because of this:

* https://github.com/jorgecarleitao/arrow2/issues/1430

Reply:
Actually, it could be the docstring of `.len()` that is wrong. I suddenly don't trust `arrow2` anymore…

Reply:
Better add a (correct!) docstring to your `fn len()` in any case